server/src/main/java/io/crate/analyze/relations/RelationAnalysisContext.java

```diff
@@ -42,8 +41,7 @@ public class RelationAnalysisContext {
     //  e.g. something like:  select * from t1, t2 must not become select t2.*, t1.*
     private final Map<RelationName, AnalyzedRelation> sources = new LinkedHashMap<>();
 
-    @Nullable
-    private List<JoinPair> joinPairs;
+    private final List<Symbol> joinConditions = new ArrayList<>();
```

Comment:
Are these even still needed?

Reply:
Yes, symbols can be present in a join condition but not referenced in the select clause.